### PR TITLE
WIP Top Stories

### PR DESCRIPTION
**Pull request by annatraussnig**
Screenshot 2020-01-28 at 18 57 29

Screenshot 2020-01-28 at 18 57 04

### TODOS

- Top story pick design/link (@garamond check my last commit for WIP on that)
- Handle corner cases (no photo, no title, file)

### CODE REVIEW

File: src/components/Discussion/Tree/CommentList.js [hunk 1]
```diff
@@ -226,21 +264,28 @@ const CommentNode = ({ t, comment }) => {
    */
   const drawLineEnd = false
 
-  const rootStyle = styles.root({ isExpanded, nestLimitExceeded })
+  const rootStyle = styles.root({ isExpanded, nestLimitExceeded, depth, board })
+  const verticalToggleStyle = isRoot
+    ? null
```
Comment (tpreusse) [inline src/components/Discussion/Tree/CommentList.js]: this is responsible for the regression with wired buttons above the name:
https://piratinnen.slack.com/archives/CSCC4NSJD/p1580830696006600
when `isRoot` both buttons where `verticalToggleStyle` is applied should not be rendered

File: src/components/Discussion/Composer/CommentComposer.js [hunk 2]
```diff
@@ -102,22 +130,37 @@ export const CommentComposer = props => {
...
+      .catch(() => {
+        if (textRef.current === text) {
+          setPreview(null)
+        }
+      })
+  }
+
+  React.useEffect(() => {
+    fetchPreview(text)
+  }, [text])
```
Comment (tpreusse) [inline src/components/Discussion/Composer/CommentComposer.js]: We should probably debounce this by 100ms or so. See `republik-frontend/components/FontSize/Sync.js` for an example.

File: src/components/Discussion/Internal/docs.md [hunk 3]
```diff
@@ -6,6 +6,26 @@ Components on this page are not exported from the styleguide. This documentation
 
 # Comment
 
+#### Link Preview
+
+_TODO: put further down that list_
```
Comment (tpreusse) [inline src/components/Discussion/Internal/docs.md]: rm or do it

File: src/components/Discussion/Tree/CommentList.js [hunk 4]
```diff
@@ -284,20 +339,39 @@ const CommentNode = ({ t, comment }) => {
...
-              dispatch({ showReplyComposer: {} })
-            }}
+            onExpand={
+              board &&
+              (() => {
+                actions.fetchMoreComments({ parentId: id, after: {} })
+              })
+            }
+            onReply={
+              !board &&
```
Comment (tpreusse) [inline src/components/Discussion/Tree/CommentList.js]: This is for a long standing issue that guests see the reply icon but it explodes when pressed:
```suggestion
              !board && !!displayAuthor &&
```

File: src/components/Discussion/Internal/Comment/Actions.js [hunk 5]
```diff
@@ -137,6 +171,18 @@ export const Actions = ({ t, comment, onReply, onEdit }) => {
...
+          onClick={onExpand}
+          title={t('styleguide/CommentActions/expand')}
+        >
+          <CommentCountIcon
+            count={comment.comments && comment.comments.totalCount}
+            small={true}
+          />
+        </IconButton>
+      )}
       {onReply && (
```
Comment (tpreusse) [inline src/components/Discussion/Internal/Comment/Actions.js]: sorry wrong location before, here is the right one:
```suggestion
      {onReply && !!displayAuthor && (
```
(for a long standing issue that guests see the reply icon)

File: src/components/Discussion/Internal/docs.md (hunk 3 above)
Comment (garamond) [reply src/components/Discussion/Internal/docs.md]: I'm not sure what it means…

Comment (tpreusse) [reply src/components/Discussion/Internal/docs.md]: I think she wanted to move it down in the doc file – first show the more common cases – e.g. move it after the body.

**Pull request merged**